src/document/mod.rs

```diff
@@ -46,14 +40,46 @@ impl ExplicitGrant {
     }
 }
 
+/// Parameters that can be provided when encrypting a new document.
+///
+/// If no document ID is provided, one will be generated for it. If no document name is provided, the document's
+/// name will be left empty.
+///
+/// For default parameters, use `DocumentEncryptOpts::default()`.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct DocumentEncryptOpts {
+    id: Option<DocumentId>,
+    name: Option<DocumentName>,
+    // at least one user/group must be included either explicitly or via a policy
+    grants: EitherOrBoth<ExplicitGrant, PolicyGrant>,
+}
+
 impl DocumentEncryptOpts {
+    /// Constructs a new `DocumentEncryptOpts`.
+    ///
+    /// This requires either an `ExplicitGrant`, a `PolicyGrant`, or both. If only using one type
+    /// of grant, consider using [with_explicit_grants()](./struct.DocumentEncryptOpts.html#method.with_explicit_grants)
+    /// or [with_policy_grants()](./struct.DocumentEncryptOpts.html#method.with_policy_grants) instead.
+    ///
+    /// # Arguments
+    /// - `id` - Unique ID to use for the document. Note: this ID will **not** be encrypted.
+    /// - `name` - Non-unique name to use for the document. Note: this name will **not** be encrypted.
+    /// - `grants` - Grants that control who will be granted access to this encrypted document.
```

Comment:
```suggestion
    /// - `grants` - Grants that control who will be granted access to decrypt this document.
```

Reply:
I worded it this way to try to imply that you're granted a little more than just the ability to decrypt (like seeing the ID, when it was last updated, who else has access, etc.). But you could probably argue that that extra information isn't worth implying. Thoughts?

Reply:
I suppose we could change it to say "who will have access to read and decrypt this document" or something like that, maybe?

Reply:
:+1: